StreamingKit/StreamingKit/STKHTTPDataSource.m

```diff
@@ -356,6 +363,14 @@ -(BOOL) parseHttpHeader
         return NO;
     }
     
+    if (redirectUrl) {
+        STKURLProvider urlProvider = ^NSURL* { return redirectUrl; };
```

Comment:
Is this a super subtle capture of self here?
self owns asyncUrlProvider, which captures self in the urlProvider?